fahrschein/src/main/java/org/zalando/fahrschein/AuthorizationBuilder.java

```diff
@@ -0,0 +1,66 @@
+package org.zalando.fahrschein;
+
+import org.zalando.fahrschein.domain.Authorization;
+import org.zalando.fahrschein.domain.Authorization.AuthorizationAttribute;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Arrays.asList;
+
+public class AuthorizationBuilder {
+
+    private final List<AuthorizationAttribute> admins;
+    private final List<AuthorizationAttribute> readers;
+
+    private AuthorizationBuilder() {
+        admins = null;
```

Comment:
Shouldn't those be empty lists?

Reply:
I am afraid that with the empty lists you will get "zero admins" instead of "authorization not set". Maybe Nakadi API experts can clarify?

Reply:
Tried it, you're right!
```
[422]: [Unprocessable Entity] [Field "authorization.readers" may not be null
```

Whole `authorization` can be `null` (meaning no restrictions), but **if** it's there — then all the fields have to be set.

Will fix.

Reply:
Hm, but actually.... it can't be an empty array either!
```
[422]: [Unprocessable Entity] [Field "authorization.readers" must contain at least one attribute
```

What do you think: should I just rewrite the builder then, so it's impossible to create an instance no providing all the needed info? Or do we want to leave it like that and just let clients get errors from the API if they use it wrong? I think it's better to rewrite not to allow improper requests at all.

Reply:
Or maybe just set defaults to `ANYONE`? Guys, what do you think? This is a matter of opinions.

Reply:
Implemented the behaviour that it's `*/*` by default and you can only override it with builder methods. Added a test and javadoc.

Reply:
> Implemented the behaviour that it's `*/*` by default and you can only override it with builder methods. Added a test and javadoc.

Sounds good.